assets/js/modules/analytics-4/components/widgets/ConnectGA4CTATileWidget.js

```diff
@@ -21,58 +21,26 @@
  */
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
-import Data from 'googlesitekit-data';
-import AnalyticsIcon from '../../../../../svg/graphics/analytics.svg';
 import ConnectModuleCTATile from '../../../../components/KeyMetrics/ConnectModuleCTATile';
-import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
-import {
-	CORE_USER,
-	keyMetricsGA4Widgets,
-} from '../../../../googlesitekit/datastore/user/constants';
-
-const { useSelect } = Data;
-
-export default function ConnectGA4CTATileWidget( { Widget, WidgetNull } ) {
-	const isGA4ModuleConnected = useSelect( ( select ) =>
-		select( CORE_MODULES ).isModuleConnected( 'analytics-4' )
-	);
-
-	const keyMetrics = useSelect( ( select ) => {
-		if ( isGA4ModuleConnected !== false ) {
-			return;
-		}
-		return select( CORE_USER ).getKeyMetrics();
-	} );
-
-	let hideWidget = true;
-
-	if ( keyMetrics?.length > 0 ) {
-		const kmAnalyticsWidgetCount = keyMetrics.filter( ( keyMetric ) =>
-			keyMetricsGA4Widgets.includes( keyMetric )
-		).length;
+import useWidgetStateEffect from '../../../../googlesitekit/widgets/hooks/useWidgetStateEffect';
 
-		if ( kmAnalyticsWidgetCount > 0 && kmAnalyticsWidgetCount < 3 ) {
-			hideWidget = false;
-		}
-	}
+export default function ConnectGA4CTATileWidget( { widgetSlug } ) {
+	// Note: `analytics` is used as the slug here since GA4 "depends" on it.
+	const metadata = useMemo( () => ( { moduleSlug: 'analytics' } ), [] );
```

Comment:
@techanvil what is the point of using the `useMemo` hook here? Is it needed only for having the same reference to the metadata object? Then we can move it out of the component and we will have the same effect without using `useMemo`, right?

```js
const metadata = { moduleSlug: '...' };

export default function ConnectGA4CTATileWidget( { widgetSlug } ) {
    useWidgetStateEffect( widgetSlug, ConnectModuleCTATile, metadata );
    ...
}
```

Reply:
Yes, I think @techanvil is following precedent in the plugin where we use that for computing a stable `metadata` object where it's necessary due to the use of a prop in the object. In this case, the object is essentially static so for now it would be fine to define it outside of the component as you've suggested 👍

Reply:
Yep - good spot there @eugene-manuilov, this is definitely a case where the memo is unneeded :+1: